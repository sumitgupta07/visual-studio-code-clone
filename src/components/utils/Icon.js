import React from 'react';
import './Icon.css';

const Icon = ({ label, onSelectIcon, iconClass }) => {
  return (
    <button onClick={() => onSelectIcon(label)}>
      <div className="tooltip">
        <div className="icon">{<i className={iconClass}></i>}</div>
        <div className="tooltiptext">
          <span>{label}</span>
        </div>
      </div>
    </button>
  );
};

export default Icon;
