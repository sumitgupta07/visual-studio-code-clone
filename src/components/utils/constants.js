export const labels = {
  FileExplorer: 'File Explorer',
  Search: 'Search',
  SourceControl: 'Source Control',
  Debug: 'Debug',
  Extensions: 'Extensions'
};

export const classNames = {
  explorerClass: 'far fa-copy fa-3x',
  searchClass: 'fas fa-search fa-rotate fa-3x',
  sourcecontrolClass: 'fas fa-code-branch fa-3x',
  debugClass: 'fas fa-bug fa-3x',
  extensionsClass: 'fas fa-th-large fa-3x'
};

export const folders = [
  {
    id: 1,
    name: 'src',
    files: [
      {
        id: 'a',
        fileName: 'index.js',
        content: 'Placeholder for index.js. It has App.js as child components'
      },
      {
        id: 'b',
        fileName: 'App.js',
        content:
          'Placeholder for App.js. It has Header and Content as child components'
      },
      {
        id: 'c',
        fileName: 'Content.js',
        content:
          'Placeholder for Content.js file. It has Navbar, SelectPane and EditorPane as child components'
      }
    ]
  }
];
