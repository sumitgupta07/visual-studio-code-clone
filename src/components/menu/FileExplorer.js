import React from 'react';
import PropTypes from 'prop-types';
import { folders } from '../utils/constants';

const FileExplorer = ({ onSelectFile }) => {
  return (
    <div className="navbar">
      <h3>File Explorer</h3>
      <i className="fas fa-folder-open">
        {'  '}
        <span>{folders[0].name}</span>
        <ul>
          {folders[0].files.map(file => (
            <li id={file.id} key={file.id} onClick={() => onSelectFile(file)}>
              <i className="far fa-file"> {file.fileName}</i>
            </li>
          ))}
        </ul>
      </i>
    </div>
  );
};

FileExplorer.propTypes = {
  onSelectFile: PropTypes.func.isRequired
};
export default FileExplorer;
