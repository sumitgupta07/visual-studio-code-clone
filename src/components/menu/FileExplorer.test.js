import React from 'react';
import { shallow } from 'enzyme';
import FileExplorer from './FileExplorer';
import { folders } from '../utils/constants';

let handleSelectFile;
let wrapper;
let mockProps;
beforeEach(() => {
  handleSelectFile = jest.fn();
  mockProps = {
    onSelectFile: handleSelectFile
  };
  wrapper = shallow(<FileExplorer {...mockProps} />);
});

it('should render File Explorer component', () => {
  expect(wrapper).toMatchSnapshot();
});
it('should call onSelectFile prop when file name is clicked', () => {
  wrapper.find('#a').simulate('click');
  expect(handleSelectFile).toHaveBeenCalled();
});

it('should all files names of first folder', () => {
  expect(wrapper.find('li').length).toEqual(folders[0].files.length);
});
