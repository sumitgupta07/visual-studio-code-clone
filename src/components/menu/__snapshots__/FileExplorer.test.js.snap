// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render File Explorer component 1`] = `
<div
  className="navbar"
>
  <h3>
    File Explorer
  </h3>
  <i
    className="fas fa-folder-open"
  >
      
    <span>
      src
    </span>
    <ul>
      <li
        id="a"
        key="a"
        onClick={[Function]}
      >
        <i
          className="far fa-file"
        >
           
          index.js
        </i>
      </li>
      <li
        id="b"
        key="b"
        onClick={[Function]}
      >
        <i
          className="far fa-file"
        >
           
          App.js
        </i>
      </li>
      <li
        id="c"
        key="c"
        onClick={[Function]}
      >
        <i
          className="far fa-file"
        >
           
          Content.js
        </i>
      </li>
    </ul>
  </i>
</div>
`;
